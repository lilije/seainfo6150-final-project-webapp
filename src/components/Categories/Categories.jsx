import React from 'react';
import styles from './Categories.module.css';

class Categories extends React.Component {
    
    displayPosts = () => {
        const listItems = this.props.posts.map((post, index) => {
            if (post.data.category === this.props.category) {
              return (
                <a className={styles.post} href={`/post/${post.id}`} key={index}>
                    <img src={post.data.imageUrl} alt="post"/>
                    <div className={styles.bottom}>
                        <h2>{post.data.title}</h2>
                        <div>
                            <span className={styles.author}>By {post.data.author}</span>
                            <span className={styles.price}>${post.data.price}</span>
                        </div>
                    </div>
                </a>
              )
            }
        
          });
          return (
              <div className={styles.root}>
                 {listItems}
              </div>
          );
    };

    render() {
        return (<div className={styles.top}>{this.displayPosts()}</div>);
    }
}

export default Categories;

