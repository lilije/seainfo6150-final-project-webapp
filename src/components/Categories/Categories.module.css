.root {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 0 auto;
  padding-bottom: 80px;
  max-width: 1500px;
}

.post {
  width: 100%;
  border-radius: 30px;
  display: flex;
  flex-direction: column;
  text-decoration: none;
  margin-bottom: 30px;
  box-shadow: 0 6px 10px rgba(0, 0, 0, 0.2);
  cursor: pointer;
}

.top {
  margin-top: 100px;
}

.price {
  color: #FF9800;
  font-weight: 600;
  margin-left: 60px;
}

h2 { color: #293963; }


img {
  width: 100%;
  max-width: 100%;
  border-top-left-radius: 30px;
  border-top-right-radius: 30px;
}

.author {
  color: #4AE6CE;
  font-weight: 600;
}

.bottom {
  display: flex;
  flex-direction: column;
  padding: 0 30px 30px;
  justify-content: space-between;
  height: 100%;
}

@media screen and (min-width: 800px) {
  .post {
    width: 48%;
  }
}

@media screen and (min-width: 1200px) {
  .post {
    width: 32%;
  }
}