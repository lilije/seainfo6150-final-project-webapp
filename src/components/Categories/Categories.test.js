import React from "react";
import { render } from "@testing-library/react";
import Categories from "./Categories.jsx";

it("renders the Categories component correctly", () => {
  const posts = [ {
      data: {
        author: "lili",
        category: "taco",
        description: "We've heard of smoke getting in your eyes, or smoke in the water, but smoky tacos? Now we've heard everything! But why not? A hint of paprika (the smoky!) on the pork, a bit of jalape√±o ranch from the slaw, and fresh pico combine for handheld dinner delight. This meal is smokin'!",
        imageUrl: "https://homechef.imgix.net/https%3A%2F%2Fasset.homechef.com%2Fuploads%2Fmeal%2Fplated%2F11888%2F419362.002.01SpanishPorkTacos_Ecomm_copy.jpg?ixlib=rails-1.1.0&w=850&auto=format&s=eadb858afc60591a00b0df808ee2a2b3",
        price: 6,
        title: "Smoky Pork Tacos"
      },
      id: "2VQc41XP6ewfAyfN2HZ93n"
    }
  ]
  const category = "taco";
  const { container } = render(<Categories posts={posts} category={category}/>);
  expect(container).toMatchSnapshot();
});