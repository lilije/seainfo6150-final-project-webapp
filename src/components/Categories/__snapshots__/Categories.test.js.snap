// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Categories component correctly 1`] = `
<div>
  <div
    class="top"
  >
    <div
      class="root"
    >
      <a
        class="post"
        href="/post/2VQc41XP6ewfAyfN2HZ93n"
      >
        <img
          alt="post"
          src="https://homechef.imgix.net/https%3A%2F%2Fasset.homechef.com%2Fuploads%2Fmeal%2Fplated%2F11888%2F419362.002.01SpanishPorkTacos_Ecomm_copy.jpg?ixlib=rails-1.1.0&w=850&auto=format&s=eadb858afc60591a00b0df808ee2a2b3"
        />
        <div
          class="bottom"
        >
          <h2>
            Smoky Pork Tacos
          </h2>
          <div>
            <span
              class="author"
            >
              By 
              lili
            </span>
            <span
              class="price"
            >
              $
              6
            </span>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`renders the FollowUs component correctly 1`] = `
<div>
  <div
    class="top"
  >
    <div
      class="root"
    >
      <a
        class="post"
        href="/post/2VQc41XP6ewfAyfN2HZ93n"
      >
        <img
          alt="post"
          src="https://homechef.imgix.net/https%3A%2F%2Fasset.homechef.com%2Fuploads%2Fmeal%2Fplated%2F11888%2F419362.002.01SpanishPorkTacos_Ecomm_copy.jpg?ixlib=rails-1.1.0&w=850&auto=format&s=eadb858afc60591a00b0df808ee2a2b3"
        />
        <div
          class="bottom"
        >
          <h2>
            Smoky Pork Tacos
          </h2>
          <div>
            <span
              class="author"
            >
              By 
              lili
            </span>
            <span
              class="price"
            >
              $
              6
            </span>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
